File: app/src/main/java/com/soundscore/app/ui/screens/ProfileScreen.kt
package com.soundscore.app.ui.screens

import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.PaddingValues
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.foundation.lazy.LazyRow
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.outlined.Download
import androidx.compose.material.icons.outlined.History
import androidx.compose.material.icons.outlined.Settings
import androidx.compose.material.icons.outlined.Share
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.getValue
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Brush
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.dp
import androidx.lifecycle.compose.collectAsStateWithLifecycle
import androidx.lifecycle.viewmodel.compose.viewModel
import androidx.compose.foundation.lazy.items
import com.soundscore.app.data.model.Album
import com.soundscore.app.data.model.UserProfile
import com.soundscore.app.ui.components.AlbumArtwork
import com.soundscore.app.ui.components.AvatarCircle
import com.soundscore.app.ui.components.BlueButton
import com.soundscore.app.ui.components.EmptyState
import com.soundscore.app.ui.components.GlassCard
import com.soundscore.app.ui.components.GlassIconButton
import com.soundscore.app.ui.components.ScreenHeader
import com.soundscore.app.ui.components.SectionHeader
import com.soundscore.app.ui.components.StatPill
import com.soundscore.app.ui.components.SyncBanner
import com.soundscore.app.ui.theme.AccentAmber
import com.soundscore.app.ui.theme.AccentGreen
import com.soundscore.app.ui.theme.AccentViolet
import com.soundscore.app.ui.theme.AlbumColors
import com.soundscore.app.ui.theme.ChromeLight
import com.soundscore.app.ui.theme.FeedItemBorder
import com.soundscore.app.ui.theme.GlassBg
import com.soundscore.app.ui.theme.GlassBorder
import com.soundscore.app.ui.theme.TextSecondary
import com.soundscore.app.ui.theme.TextTertiary
import com.soundscore.app.ui.viewmodel.ProfileUiState
import com.soundscore.app.ui.viewmodel.ProfileViewModel

@Composable
fun ProfileScreen(
    modifier: Modifier = Modifier,
    profileViewModel: ProfileViewModel = viewModel(),
) {
    val uiState by profileViewModel.uiState.collectAsStateWithLifecycle()
    ProfileScreenContent(
        uiState = uiState,
        modifier = modifier,
    )
}

@Composable
fun ProfileScreenContent(
    uiState: ProfileUiState,
    modifier: Modifier = Modifier,
) {
    val profile = uiState.profile

    if (profile == null) {
        Box(modifier = modifier.fillMaxSize(), contentAlignment = Alignment.Center) {
            Text("Loading profile...", color = TextSecondary)
        }
        return
    }

    LazyColumn(
        modifier = modifier.fillMaxSize(),
        contentPadding = PaddingValues(start = 20.dp, top = 16.dp, end = 20.dp, bottom = 120.dp),
        verticalArrangement = Arrangement.spacedBy(16.dp),
    ) {
        item {
            SyncBanner(message = uiState.syncMessage)
        }

        item {
            ProfileHeader(profile = profile)
        }

        item {
            Row(
                modifier = Modifier.fillMaxWidth(),
                horizontalArrangement = Arrangement.spacedBy(10.dp),
            ) {
                StatPill(
                    value = "${profile.albumsCount}",
                    label = "Albums",
                    modifier = Modifier.weight(1f),
                    highlight = true,
                )
                StatPill(
                    value = "${profile.reviewCount}",
                    label = "Reviews",
                    modifier = Modifier.weight(1f),
                )
                StatPill(
                    value = "${profile.listCount}",
                    label = "Lists",
                    modifier = Modifier.weight(1f),
                )
                StatPill(
                    value = String.format("%.1f", profile.avgRating),
                    label = "Avg",
                    modifier = Modifier.weight(1f),
                    highlight = true,
                    accentColor = AccentAmber,
                )
            }
        }

        item {
            Row(
                modifier = Modifier.fillMaxWidth(),
                horizontalArrangement = Arrangement.SpaceEvenly,
            ) {
                GlassIconButton(
                    icon = Icons.Outlined.Share,
                    label = "Share",
                    tint = AccentGreen,
                )
                GlassIconButton(
                    icon = Icons.Outlined.Download,
                    label = "Export",
                )
                GlassIconButton(
                    icon = Icons.Outlined.Settings,
                    label = "Settings",
                )
            }
        }

        if (uiState.favoriteAlbums.isNotEmpty()) {
            item {
                SectionHeader(eyebrow = "Favorites", title = "Pinned to your identity")
            }

            item {
                FavoriteGrid(albums = uiState.favoriteAlbums)
            }
        }

        item {
            SectionHeader(eyebrow = "Taste DNA", title = "Genres on repeat")
        }

        item {
            TasteTags(tags = profile.genres)
        }

        if (uiState.latestRecap != null) {
            item {
                SectionHeader(eyebrow = "Weekly recap", title = "Your week in music")
            }

            item {
                RecapCard(
                    totalLogs = uiState.latestRecap!!.totalLogs,
                    avgRating = uiState.latestRecap!!.averageRating,
                    shareText = uiState.latestRecap!!.shareText,
                )
            }
        }

        item {
            SectionHeader(eyebrow = "Activity", title = "Recent ratings")
        }

        items(uiState.recentActivity) { item ->
            GlassCard(
                cornerRadius = 16.dp,
                borderColor = FeedItemBorder,
                contentPadding = PaddingValues(10.dp),
            ) {
                Row(
                    modifier = Modifier.fillMaxWidth(),
                    horizontalArrangement = Arrangement.spacedBy(10.dp),
                    verticalAlignment = Alignment.CenterVertically,
                ) {
                    AlbumArtwork(
                        artworkUrl = item.album.artworkUrl,
                        colors = item.album.artColors,
                        modifier = Modifier.size(44.dp),
                        cornerRadius = 12.dp,
                    )
                    Column(modifier = Modifier.weight(1f)) {
                        Text(
                            text = item.album.title,
                            style = MaterialTheme.typography.titleMedium,
                            fontWeight = FontWeight.SemiBold,
                        )
                        Text(
                            text = item.action,
                            style = MaterialTheme.typography.bodySmall,
                            color = TextSecondary,
                        )
                    }
                    Text(
                        text = item.timeAgo,
                        style = MaterialTheme.typography.labelSmall,
                        color = TextTertiary,
                    )
                }
            }
        }

        if (uiState.recentActivity.isEmpty()) {
            item {
                EmptyState(
                    title = "Recent activity",
                    subtitle = "Your latest ratings and reviews will appear here.",
                    icon = Icons.Outlined.History,
                )
            }
        }
    }
}

@Composable
private fun ProfileHeader(profile: UserProfile) {
    GlassCard(
        cornerRadius = 26.dp,
        borderColor = FeedItemBorder,
        frosted = true,
    ) {
        Column(
            modifier = Modifier.fillMaxWidth(),
            horizontalAlignment = Alignment.CenterHorizontally,
        ) {
            AvatarCircle(
                initials = profile.handle.removePrefix("@").take(2),
                gradientColors = listOf(AccentGreen, AccentViolet),
                size = 80.dp,
            )
            Spacer(Modifier.height(12.dp))
            Text(
                text = profile.handle,
                style = MaterialTheme.typography.headlineMedium,
                color = ChromeLight,
                fontWeight = FontWeight.Bold,
            )
            Spacer(Modifier.height(4.dp))
            Text(
                text = profile.bio,
                style = MaterialTheme.typography.bodyMedium,
                color = TextSecondary,
                textAlign = TextAlign.Center,
            )
            Spacer(Modifier.height(12.dp))
            Row(
                horizontalArrangement = Arrangement.spacedBy(24.dp),
            ) {
                ProfileCount(value = "${profile.followingCount}", label = "Following")
                ProfileCount(value = "${profile.followersCount}", label = "Followers")
            }
        }
    }
}

@Composable
private fun ProfileCount(value: String, label: String) {
    Column(horizontalAlignment = Alignment.CenterHorizontally) {
        Text(
            text = value,
            style = MaterialTheme.typography.titleLarge,
            color = ChromeLight,
            fontWeight = FontWeight.Bold,
        )
        Text(
            text = label,
            style = MaterialTheme.typography.labelSmall,
            color = TextTertiary,
        )
    }
}

@Composable
private fun FavoriteGrid(albums: List<Album>) {
    Column(verticalArrangement = Arrangement.spacedBy(10.dp)) {
        albums.chunked(3).forEach { rowAlbums ->
            Row(
                modifier = Modifier.fillMaxWidth(),
                horizontalArrangement = Arrangement.spacedBy(10.dp),
            ) {
                rowAlbums.forEach { album ->
                    GlassCard(
                        modifier = Modifier.weight(1f),
                        fillMaxWidth = true,
                        cornerRadius = 18.dp,
                        borderColor = FeedItemBorder,
                        contentPadding = PaddingValues(6.dp),
                        onClick = { },
                    ) {
                        Column(verticalArrangement = Arrangement.spacedBy(6.dp)) {
                            AlbumArtwork(
                                artworkUrl = album.artworkUrl,
                                colors = album.artColors,
                                modifier = Modifier
                                    .fillMaxWidth()
                                    .height(100.dp),
                                cornerRadius = 14.dp,
                            )
                            Text(
                                text = album.title,
                                style = MaterialTheme.typography.titleSmall,
                                maxLines = 1,
                                fontWeight = FontWeight.Medium,
                            )
                        }
                    }
                }
                repeat(3 - rowAlbums.size) {
                    Spacer(modifier = Modifier.weight(1f))
                }
            }
        }
    }
}

@Composable
private fun TasteTags(tags: List<String>) {
    LazyRow(
        horizontalArrangement = Arrangement.spacedBy(8.dp),
    ) {
        items(tags.size) { index ->
            val tag = tags[index]
            val tagColors = when (index % 4) {
                0 -> AlbumColors.orchid
                1 -> AlbumColors.lagoon
                2 -> AlbumColors.ember
                else -> AlbumColors.rose
            }
            Box(
                modifier = Modifier
                    .clip(RoundedCornerShape(20.dp))
                    .background(
                        Brush.linearGradient(
                            listOf(tagColors.first().copy(alpha = 0.4f), tagColors.last().copy(alpha = 0.15f))
                        )
                    )
                    .border(0.5.dp, tagColors.last().copy(alpha = 0.3f), RoundedCornerShape(20.dp))
                    .padding(horizontal = 14.dp, vertical = 8.dp),
            ) {
                Text(
                    text = tag,
                    style = MaterialTheme.typography.labelMedium,
                    color = ChromeLight,
                    fontWeight = FontWeight.Medium,
                )
            }
        }
    }
}

@Composable
private fun RecapCard(
    totalLogs: Int,
    avgRating: Float,
    shareText: String,
) {
    GlassCard(
        cornerRadius = 22.dp,
        tintColor = AccentGreen,
        borderColor = AccentGreen.copy(alpha = 0.2f),
    ) {
        Column(verticalArrangement = Arrangement.spacedBy(10.dp)) {
            Row(
                modifier = Modifier.fillMaxWidth(),
                horizontalArrangement = Arrangement.SpaceBetween,
            ) {
                Column {
                    Text(
                        text = "$totalLogs",
                        style = MaterialTheme.typography.headlineMedium,
                        color = AccentGreen,
                        fontWeight = FontWeight.Black,
                    )
                    Text(
                        text = "ALBUMS LOGGED",
                        style = MaterialTheme.typography.labelSmall,
                        color = TextTertiary,
                    )
                }
                Column(horizontalAlignment = Alignment.End) {
                    Text(
                        text = String.format("%.1f", avgRating),
                        style = MaterialTheme.typography.headlineMedium,
                        color = AccentAmber,
                        fontWeight = FontWeight.Black,
                    )
                    Text(
                        text = "AVG RATING",
                        style = MaterialTheme.typography.labelSmall,
                        color = TextTertiary,
                    )
                }
            }
            Text(
                text = shareText,
                style = MaterialTheme.typography.bodyMedium,
                color = TextSecondary,
            )
            Row(horizontalArrangement = Arrangement.spacedBy(10.dp)) {
                BlueButton(text = "View Recap", onClick = { })
                GlassIconButton(
                    icon = Icons.Outlined.Share,
                    label = "Share",
                    tint = AccentGreen,
                )
            }
        }
    }
}

File: ios/SoundScore/SoundScore/Screens/ProfileScreen.swift
import SwiftUI

struct ProfileScreen: View {
    @StateObject private var viewModel = ProfileViewModel()
    var onSelectAlbum: (Album) -> Void = { _ in }
    var onOpenSettings: () -> Void = {}
    @State private var appeared = false

    var body: some View {
        ScrollView {
            LazyVStack(alignment: .leading, spacing: 16) {
                SyncBanner(message: viewModel.syncMessage)

                if let error = viewModel.errorMessage {
                    ErrorBanner(message: error)
                }

                heroBanner

                if !viewModel.favoriteAlbums.isEmpty {
                    favoritesSection
                }

                if viewModel.tasteDNA != nil {
                    tasteDNASection
                }

                if let recap = viewModel.recap {
                    recapCard(recap)
                }

                if !viewModel.recentActivity.isEmpty {
                    recentActivitySection
                }
            }
            .padding(.bottom, 120)
        }
        .refreshable { await SoundScoreRepository.shared.refresh() }
        .onAppear { appeared = true }
    }

    // MARK: - Hero Banner

    private var heroBanner: some View {
        ZStack(alignment: .bottom) {
            ZStack {
                let artworks = viewModel.favoriteAlbums.prefix(4)
                if !artworks.isEmpty {
                    GeometryReader { geo in
                        let size = geo.size
                        ZStack {
                            ForEach(Array(artworks.enumerated()), id: \.element.id) { index, album in
                                AlbumArtwork(artworkUrl: album.artworkUrl, colors: album.artColors, cornerRadius: 0)
                                    .frame(width: size.width / 2, height: 140)
                                    .offset(
                                        x: index % 2 == 0 ? -size.width / 4 : size.width / 4,
                                        y: index < 2 ? -70 : 70
                                    )
                            }
                        }
                        .frame(width: size.width, height: size.height)
                        .blur(radius: 30)
                    }
                } else {
                    ThemeManager.shared.primary.opacity(0.15)
                }

                LinearGradient(
                    colors: [SSColors.overlayMedium, SSColors.overlayDark],
                    startPoint: .top, endPoint: .bottom
                )
            }
            .frame(height: 300)

            VStack(spacing: 10) {
                ZStack {
                    Circle()
                        .fill(ThemeManager.shared.primary.opacity(0.2))
                        .frame(width: 106, height: 106)
                        .blur(radius: 12)

                    AvatarCircle(
                        initials: String(viewModel.handle.dropFirst().prefix(2)),
                        gradientColors: [ThemeManager.shared.primary, SSColors.accentViolet],
                        size: 96
                    )
                    .overlay(
                        Circle()
                            .stroke(ThemeManager.shared.primary, lineWidth: 3)
                            .frame(width: 96, height: 96)
                    )
                    .shadow(color: ThemeManager.shared.primary.opacity(0.4), radius: 12, y: 4)
                }

                Text(viewModel.handle)
                    .font(SSTypography.displayMedium)
                    .foregroundColor(SSColors.chromeLight)
                    .fontWeight(.bold)

                Text(viewModel.bio)
                    .font(SSTypography.bodyMedium)
                    .foregroundColor(SSColors.textSecondary)
                    .multilineTextAlignment(.center)
                    .padding(.horizontal, 32)

                // Inline stats
                inlineStatsText()
                    .lineLimit(1)
                    .minimumScaleFactor(0.75)
                    .padding(.horizontal, 24)
            }
            .padding(.bottom, 20)
        }
        .overlay(alignment: .topTrailing) {
            Button { onOpenSettings() } label: {
                Image(systemName: "gearshape")
                    .font(.system(size: 15, weight: .medium))
                    .foregroundColor(SSColors.chromeLight)
                    .frame(width: 40, height: 40)
                    .background(.ultraThinMaterial)
                    .clipShape(Circle())
                    .overlay(Circle().stroke(SSColors.glassBorder, lineWidth: 0.5))
            }
            .buttonStyle(.plain)
            .padding(.top, 52)
            .padding(.trailing, 16)
        }
    }

    private func inlineStatsText() -> Text {
        let items = viewModel.metrics
        var result = Text("")
        for (index, metric) in items.enumerated() {
            if index > 0 {
                result = result + Text("  \u{00B7}  ")
                    .font(SSTypography.bodySmall)
                    .foregroundColor(SSColors.chromeDim)
            }
            result = result
                + Text(metric.value)
                    .font(SSTypography.bodySmall)
                    .fontWeight(.bold)
                    .foregroundColor(ThemeManager.shared.primary)
                + Text(" \(metric.label.lowercased())")
                    .font(SSTypography.bodySmall)
                    .foregroundColor(SSColors.textTertiary)
        }
        return result
    }

    // MARK: - Favorites

    private var favoritesSection: some View {
        VStack(alignment: .leading, spacing: 10) {
            SectionHeader(eyebrow: "Top picks", title: "Favorites", trailing: "See All")
                .padding(.horizontal, 20)

            ScrollView(.horizontal, showsIndicators: false) {
                LazyHStack(spacing: 12) {
                    ForEach(Array(viewModel.favoriteAlbums.enumerated()), id: \.element.id) { index, album in
                        Button {
                            UIImpactFeedbackGenerator(style: .light).impactOccurred()
                            onSelectAlbum(album)
                        } label: {
                            ZStack(alignment: .bottom) {
                                AlbumArtwork(artworkUrl: album.artworkUrl, colors: album.artColors, cornerRadius: 18)

                                LinearGradient(
                                    colors: [.clear, SSColors.overlayDark],
                                    startPoint: .init(x: 0.5, y: 0.4), endPoint: .bottom
                                )
                                .clipShape(RoundedRectangle(cornerRadius: 18))

                                VStack(alignment: .leading, spacing: 2) {
                                    Text(album.title)
                                        .font(SSTypography.titleMedium)
                                        .foregroundColor(SSColors.chromeLight)
                                        .fontWeight(.bold)
                                        .lineLimit(1)
                                    Text(album.artist)
                                        .font(SSTypography.labelSmall)
                                        .foregroundColor(SSColors.textSecondary)
                                        .lineLimit(1)
                                }
                                .padding(10)
                                .frame(maxWidth: .infinity, alignment: .leading)
                            }
                            .frame(width: 140, height: 180)
                            .clipShape(RoundedRectangle(cornerRadius: 18))
                            .overlay(RoundedRectangle(cornerRadius: 18).stroke(SSColors.feedItemBorder, lineWidth: 0.5))
                        }
                        .buttonStyle(.plain)
                        .opacity(appeared ? 1 : 0)
                        .offset(y: appeared ? 0 : 20)
                        .animation(.spring(response: 0.5, dampingFraction: 0.7).delay(Double(index) * 0.08), value: appeared)
                    }
                }
                .padding(.horizontal, 20)
            }
        }
    }

    // MARK: - Taste DNA

    private var tasteDNASection: some View {
        VStack(alignment: .leading, spacing: 14) {
            SectionHeader(eyebrow: "Your vibe", title: "Taste DNA")
                .padding(.horizontal, 20)

            if let dna = viewModel.tasteDNA {
                // Sound DNA summary (AI-generated 3-word tagline)
                if let summary = viewModel.soundDNASummary {
                    Text(summary)
                        .font(.system(size: 26, weight: .black, design: .rounded))
                        .foregroundStyle(
                            LinearGradient(
                                colors: [ThemeManager.shared.primary, SSColors.accentViolet],
                                startPoint: .leading, endPoint: .trailing
                            )
                        )
                        .padding(.horizontal, 20)
                        .padding(.bottom, 4)
                }

                // Top Genres — horizontal bar chart with animated entry
                VStack(alignment: .leading, spacing: 8) {
                    let palettes: [[Color]] = [
                        AlbumColors.forest, AlbumColors.orchid, AlbumColors.lagoon,
                        AlbumColors.ember, AlbumColors.rose, AlbumColors.midnight,
                    ]
                    ForEach(Array(dna.topGenres.enumerated()), id: \.element.genre) { index, entry in
                        GenreBarRow(
                            genre: entry.genre,
                            weight: entry.weight,
                            colors: palettes[index % palettes.count],
                            delay: Double(index) * 0.1
                        )
                    }
                }
                .padding(.horizontal, 20)

                // Taste Stats Row — 3 glass pills
                ScrollView(.horizontal, showsIndicators: false) {
                    HStack(spacing: 10) {
                        tasteStatPill(
                            icon: "scalemass",
                            label: dna.ratingStyle,
                            colors: AlbumColors.amber
                        )
                        if let topDecade = dna.decadeBreakdown.first {
                            tasteStatPill(
                                icon: "calendar",
                                label: "\(topDecade.decade) Native",
                                colors: AlbumColors.lagoon
                            )
                        }
                        tasteStatPill(
                            icon: "globe",
                            label: "Explorer \(String(format: "%.1f", dna.diversityScore))",
                            colors: AlbumColors.orchid
                        )
                    }
                    .padding(.horizontal, 20)
                }

                // Controversial Pick
                if let pick = dna.controversialPick {
                    GlassCard(tintColor: SSColors.accentCoral, cornerRadius: 18,
                              borderColor: SSColors.accentCoral.opacity(0.3)) {
                        VStack(alignment: .leading, spacing: 6) {
                            HStack(spacing: 6) {
                                Image(systemName: "flame.fill")
                                    .foregroundColor(SSColors.accentCoral)
                                    .font(.system(size: 14))
                                Text("Hottest Take")
                                    .font(SSTypography.labelMedium)
                                    .foregroundColor(SSColors.accentCoral)
                                    .fontWeight(.bold)
                            }
                            Text("You gave **\(pick.album)** a \(String(format: "%.1f", pick.rating)) when the community says \(String(format: "%.1f", pick.communityAvg))")
                                .font(SSTypography.bodyMedium)
                                .foregroundColor(SSColors.chromeLight)
                        }
                    }
                    .padding(.horizontal, 20)
                }
            }
        }
        .onAppear { viewModel.generateSoundDNA() }
    }

    // MARK: - Animated Genre Bar

    private struct GenreBarRow: View {
        let genre: String
        let weight: Float
        let colors: [Color]
        let delay: Double
        @State private var animatedWidth: CGFloat = 0

        var body: some View {
            HStack(spacing: 10) {
                Text(genre)
                    .font(SSTypography.labelMedium)
                    .foregroundColor(SSColors.chromeLight)
                    .frame(width: 100, alignment: .leading)

                GeometryReader { geo in
                    RoundedRectangle(cornerRadius: 6)
                        .fill(
                            LinearGradient(
                                colors: colors,
                                startPoint: .leading, endPoint: .trailing
                            )
                        )
                        .frame(width: animatedWidth)
                        .onAppear {
                            let target = max(geo.size.width * CGFloat(weight), 20)
                            withAnimation(.spring(response: 0.6, dampingFraction: 0.7).delay(delay)) {
                                animatedWidth = target
                            }
                        }
                }
                .frame(height: 20)

                Text("\(Int(weight * 100))%")
                    .font(SSTypography.labelSmall)
                    .foregroundColor(SSColors.textTertiary)
                    .frame(width: 36, alignment: .trailing)
            }
        }
    }

    private func tasteStatPill(icon: String, label: String, colors: [Color]) -> some View {
        HStack(spacing: 6) {
            Image(systemName: icon)
                .font(.system(size: 13))
                .foregroundStyle(
                    LinearGradient(colors: colors, startPoint: .topLeading, endPoint: .bottomTrailing)
                )
            Text(label)
                .font(SSTypography.labelMedium)
                .foregroundColor(SSColors.chromeLight)
                .fontWeight(.semibold)
        }
        .padding(.horizontal, 14)
        .padding(.vertical, 10)
        .background(SSColors.glassBg)
        .clipShape(Capsule())
        .overlay(Capsule().stroke(SSColors.glassBorder, lineWidth: 0.5))
    }

    // MARK: - Recap

    private func recapCard(_ recap: WeeklyRecap) -> some View {
        GlassCard(tintColor: ThemeManager.shared.primary, cornerRadius: 22, borderColor: ThemeManager.shared.primary.opacity(0.2)) {
            VStack(spacing: 12) {
                HStack {
                    Image(systemName: "chart.bar.fill")
                        .foregroundColor(ThemeManager.shared.primary)
                    Text("Weekly Recap")
                        .font(SSTypography.headlineSmall)
                        .foregroundColor(SSColors.chromeLight)
                        .fontWeight(.bold)
                    Spacer()
                }

                HStack {
                    VStack(spacing: 2) {
                        Text("\(recap.totalLogs)")
                            .font(SSTypography.headlineMedium)
                            .foregroundColor(SSColors.chromeLight)
                            .fontWeight(.black)
                        Text("LOGS")
                            .font(SSTypography.labelSmall)
                            .foregroundColor(SSColors.textTertiary)
                    }
                    .frame(maxWidth: .infinity)

                    VStack(spacing: 2) {
                        Text(String(format: "%.1f", recap.averageRating))
                            .font(SSTypography.headlineMedium)
                            .foregroundColor(SSColors.accentAmber)
                            .fontWeight(.black)
                        Text("AVG")
                            .font(SSTypography.labelSmall)
                            .foregroundColor(SSColors.textTertiary)
                    }
                    .frame(maxWidth: .infinity)
                }

                ShareLink(item: recap.shareText) {
                    HStack {
                        Image(systemName: "square.and.arrow.up")
                            .font(.system(size: 13))
                        Text("Share Recap")
                            .font(SSTypography.labelMedium)
                    }
                    .foregroundColor(ThemeManager.shared.primary)
                    .padding(.horizontal, 16)
                    .padding(.vertical, 8)
                    .background(ThemeManager.shared.primary.opacity(0.15))
                    .clipShape(Capsule())
                }
            }
        }
        .padding(.horizontal, 20)
    }

    // MARK: - Recent Activity

    private var recentActivitySection: some View {
        VStack(alignment: .leading, spacing: 10) {
            SectionHeader(eyebrow: "Recent", title: "Activity")
                .padding(.horizontal, 20)

            ForEach(viewModel.recentActivity) { entry in
                HStack(spacing: 0) {
                    RoundedRectangle(cornerRadius: 1)
                        .fill(LinearGradient(colors: entry.album.artColors, startPoint: .top, endPoint: .bottom))
                        .frame(width: 3)
                        .padding(.vertical, 4)

                    HStack(spacing: 10) {
                        AlbumArtwork(artworkUrl: entry.album.artworkUrl, colors: entry.album.artColors, cornerRadius: 12)
                            .frame(width: 56, height: 56)
                            .onTapGesture {
                                UIImpactFeedbackGenerator(style: .light).impactOccurred()
                                onSelectAlbum(entry.album)
                            }

                        VStack(alignment: .leading, spacing: 3) {
                            Text(entry.album.title)
                                .font(SSTypography.titleMedium)
                                .foregroundColor(SSColors.chromeLight)
                                .fontWeight(.semibold)
                                .lineLimit(1)
                            Text(entry.album.artist)
                                .font(SSTypography.bodySmall)
                                .foregroundColor(SSColors.textSecondary)
                                .lineLimit(1)
                        }

                        Spacer()

                        VStack(alignment: .trailing, spacing: 3) {
                            StarRating(rating: entry.rating, starSize: 10)
                            Text(entry.dateLabel)
                                .font(SSTypography.labelSmall)
                                .foregroundColor(SSColors.textTertiary)
                        }
                    }
                    .padding(.horizontal, 10)
                    .padding(.vertical, 8)
                }
                .background(SSColors.glassBg)
                .clipShape(RoundedRectangle(cornerRadius: 14))
                .overlay(RoundedRectangle(cornerRadius: 14).stroke(SSColors.feedItemBorder, lineWidth: 0.5))
                .padding(.horizontal, 20)
            }
        }
    }
}
